tests/db/test_pipetask_error_type.py

```diff
@@ -0,0 +1,137 @@
+import pytest
+from sqlalchemy.exc import IntegrityError
+from sqlalchemy.ext.asyncio import async_scoped_session
+
+from lsst.cmservice import db
+
+
+@pytest.mark.asyncio()
+async def test_error_match(session: async_scoped_session) -> None:
+    """Test error matching in pipetask_error_type.match.
+
+    Correctly match a real error to the error_type database and fail to match a
+    fake error which is not in the database.
+    """
+    # Here we list an error which we will insert into the database table.
+    known_error = {
+        "source": "manifest",
+        "flavor": "configuration",
+        "action": "review",
+        "task_name": "skyObjectMean",
+        "diagnostic_message": "The error message from a regular pipetask error",
+    }
+    # First we want to make a row of the database table for the error.
+    e1 = await db.PipetaskErrorType.create_row(
+        session,
+        source=known_error["source"],
+        flavor=known_error["flavor"],
+        action=known_error["action"],
+        task_name=known_error["task_name"],
+        diagnostic_message=known_error["diagnostic_message"],
+    )
+
+    # Assert that the error we just put in the database will match with itself
+    assert e1.match(
+        known_error["task_name"],
+        known_error["diagnostic_message"],
+    ), "Known error does not match to its value stored in the database"
+
+    # Here we test that the same error with a different pipetask name will not
+    # match
+    wrong_pipetask = {
+        "source": "manifest",
+        "flavor": "configuration",
+        "action": "review",
+        "task_name": "isr",
+        "diagnostic_message": "The error message from a regular pipetask error",
+    }
+    assert (
+        e1.match(wrong_pipetask["task_name"], wrong_pipetask["diagnostic_message"]) is False
+    ), "Failure to identify non-matching pipetask name"
+
+    # Here we test that the same pipetask name but different error does not
+    # match.
+    wrong_message = {
+        "source": "manifest",
+        "flavor": "configuration",
+        "action": "review",
+        "task_name": "skyObjectMean",
+        "diagnostic_message": "A different error message",
+    }
+    assert (
+        e1.match(wrong_message["task_name"], wrong_message["diagnostic_message"]) is False
+    ), "Failure to identify separate errors with the same associated pipetask"
```

Comment:
What do you think of using combinations of `known_error[]` for good args, and string constants at point of call for bad args, for this test and the next, as you did for the empty strings in the last test below?  Then you could dump the `wrong_` message defs, which would reduce line count and might improve readability (because reader wouldn't having to dig through the `wrong_` structures carefully to see what is same/different?)

Reply:
Hmm. I made the choice thinking it was clearer, but I think I can throw something into the strings that makes it non-confusing either way. :)